### PR TITLE
Fixed keyboard submit

### PR DESCRIPTION
**Pull request by iondrimba**
This PR includes a fix for submitting form data through keyboard `enter`. This problem started to happen after we refactored our input validations from `onFieldChange` to `onFieldBlur`. Now we have both callbacks `onFieldBlur` (still validates data) and `onFieldChange` keeps state up to date so we can submit through keyboard input.

**CHANGELOG** :memo:

* Update `onFieldChange` callback
* Added `onFieldBlur` callback

**PRINTS** :framed_picture:

### CODE REVIEW

File: src/helpers/zipcode.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const ZIPCODE_VALID_LENGTH = 8;
+const ZIPCODE_VALID_LENGTH = 9;
```
Comment (iondrimba) [inline src/helpers/zipcode.js]: Changed to validate zipcode with full format `99999-999` and only do cleaning on submit

File: src/components/Input.js [hunk 2]
```diff
@@ -63,7 +69,14 @@ export default class Input extends Component {
 
     if (type === 'phone') {
       this.mask = new IMask(this.ref.current, { mask: '(00) 00000-0000' });
-      this.mask.on('complete', () => { this.setState({ value: this.mask.value }); });
+      this.mask.on('complete', () => {
+        this.props.onFieldChange({
```
Comment (iondrimba) [inline src/components/Input.js]: Added onFieldChange to keep outer state synched

File: src/components/Form/index.js [hunk 3]
```diff
@@ -162,6 +162,18 @@ export default class Form extends Component {
     this.updateStep(fields);
   }
 
+  onFieldChange({ value, id }) {
+    const fields = this.currentStep.fields.map((item) => {
```
Comment (iondrimba) [inline src/components/Form/index.js]: We only keep state updated without validation

File: spec/components/Input.spec.js [hunk 4]
```diff
@@ -17,15 +17,20 @@ function createNodeMock(element) {
 }
 
 describe('Input', () => {
+  const commonProps = {
```
Comment (williamcosta) [inline spec/components/Input.spec.js]: Nice! 👏

**Pull request merged**